src/parser/jobs/blm/modules/Procs.js

```diff
@@ -81,35 +75,23 @@ export default class Procs extends Module {
 		this.timeline.addGroup(this._group) // Group for showing procs on the timeline
 	}
 
-	// Keep track of casts we start to help look for instant casts
-	_onBeginCast(event) {
-		this._castingSpell = event.ability
+	_onLoseProc(event) {
+		this._stopAndSave(event.ability.guid, event.timestamp)
 	}
 
-	// Consolidate old onCast functions into one central function
-	_onCast(event) {
-		// Skip proc checking if we had a corresponding begincast event or the begincast we recorded isn't the same as this spell (ie. cancelled a cast, used a proc)
-		if (!this._castingSpell || this._castingSpell !== event.ability) {
-			if (event.ability.guid === ACTIONS.FIRE_III.id) {
-				if (this._firestarter !== null) {
-					event.ability.overrideAction = ACTIONS.FIRE_III_PROC
-				}
-			} else if (THUNDER_ACTIONS.includes(event.ability.guid)) {
-				if (this._thundercloud) {
-					this.castTime.set(THUNDER_ACTIONS, 0, event.timestamp, event.timestamp) // Note that this cast was 0 time
-					if (event.ability.guid === ACTIONS.THUNDER_III.id) {
-						event.ability.overrideAction = ACTIONS.THUNDER_III_PROC // Mark this T3 as a proc for use elsewhere
-					} else if (event.ability.guid === ACTIONS.THUNDER_IV.id) {
-						event.ability.overrideAction = ACTIONS.THUNDER_IV_PROC // Might as well mark out T4 procs as well...
-					}
-				}
-			}
+	_onGainProc(event) {
+		const status = getStatus(event.ability.guid)
```

Comment:
If you literally only use the ID here, might be easier to just assign `statusId` directly without `getStatus()`.